rpc/extension/server.go

```diff
@@ -0,0 +1,93 @@
+package extension
+
+import (
+	"context"
+	"net"
+
+	"github.com/sensu/sensu-go/rpc"
+	"google.golang.org/grpc"
+)
+
+var _ rpc.ExtensionServer = &Server{}
+
+// Server is a Sensu extension.
+//
+// Create new extensions that will run on a local port with NewServer, and
+// then supply any of HandlerFunc, MutatorFunc or FilterFunc via the Handler
+// Mutator
+type Server struct {
+	ext    Interface
+	ln     net.Listener
+	server *grpc.Server
+}
+
+// NewServer creates a new server that will run on the provided listener.
+func NewServer(extension Interface, ln net.Listener) *Server {
+	return &Server{
+		ln:     ln,
+		server: grpc.NewServer(),
+		ext:    extension,
+	}
+}
+
+// Start starts the extension service.
+func (s *Server) Start() error {
+	rpc.RegisterExtensionServer(s.server, s)
+	go func() {
+		// TODO: add error handling once gRPC errors are a little less
```

Comment:
Out of curiosity, what do the gRPC errors look like now?

Reply:
I think it's this: https://github.com/grpc/grpc-go/issues/1737

Not sure, since it's been merged in.